flake8_type_checking/checker.py

```diff
@@ -360,28 +361,147 @@ def import_type(self) -> ImportTypeValue:
         return cast('ImportTypeValue', classify_base(self.full_name.partition('.')[0]))
 
 
-class UnwrappedAnnotation(NamedTuple):
-    """Represents a single `ast.Name` in an unwrapped annotation."""
+class WrappedAnnotation(NamedTuple):
+    """Represents a wrapped annotation i.e. a string constant."""
 
     lineno: int
     col_offset: int
     annotation: str
+    names: set[str]
+    scope: Scope
     type: Literal['annotation', 'alias', 'new-alias']
 
 
-class WrappedAnnotation(NamedTuple):
-    """Represents a wrapped annotation i.e. a string constant."""
+class UnwrappedAnnotation(NamedTuple):
+    """Represents a single `ast.Name` in an unwrapped annotation."""
 
     lineno: int
     col_offset: int
     annotation: str
-    names: set[str]
+    scope: Scope
     type: Literal['annotation', 'alias', 'new-alias']
 
 
+class Symbol(NamedTuple):
+    """Represents an import/definition/declaration of a variable."""
```

Comment:
@sondrelg A declaration is something like `x: Foo` i.e. something that gets stored into `__annotations__` but doesn't go into `locals`/`globals` yet because it hasn't been assigned a value.

A definition is an assignment, a function definition or a class definition, i.e. something that creates/updates the corresponding name in the locals.